File: clients/ios/Lextures/Core/LMS/LMSFeatureModelsIntroCourse.swift
import Foundation

// MARK: - Intro course (IC07)

/// Canonical intro course URL code (matches server introcourse.CourseCode).
enum IntroCourseConstants {
    static let courseCode = "C-WLCOME"
}

typealias IntroCourseModuleStatus = String

struct IntroCourseNextItem: Codable, Equatable {
    var slug: String
    var title: String
    var route: String
}

struct IntroCourseModuleProgress: Codable, Equatable, Identifiable {
    var slug: String
    var title: String
    var status: IntroCourseModuleStatus

    var id: String { slug }
}

struct IntroCourseProgress: Codable, Equatable {
    var enrolled: Bool
    var courseCode: String?
    var modulesComplete: Int
    var modulesTotal: Int
    var percent: Int
    var runningGrade: Double?
    var completedAt: String?
    var credentialId: String?
    var nextItem: IntroCourseNextItem?
    var modules: [IntroCourseModuleProgress]?
    var welcomeBannerDismissed: Bool?
    var celebrationSeen: Bool?
}

enum IntroCourseCardState: Equatable {
    case hidden
    case loading
    case error
    case notStarted
    case inProgress
    case completed
}

File: clients/android/app/src/main/kotlin/com/lextures/android/core/lms/LmsFeatureModelsIntroCourse.kt
package com.lextures.android.core.lms

import kotlinx.serialization.Serializable

/** Canonical intro course URL code (matches server introcourse.CourseCode). */
object IntroCourseConstants {
    const val courseCode = "C-WLCOME"
}

@Serializable
data class IntroCourseNextItem(
    val slug: String,
    val title: String,
    val route: String,
)

@Serializable
data class IntroCourseModuleProgress(
    val slug: String,
    val title: String,
    val status: String,
)

@Serializable
data class IntroCourseProgress(
    val enrolled: Boolean,
    val courseCode: String? = null,
    val modulesComplete: Int = 0,
    val modulesTotal: Int = 7,
    val percent: Int = 0,
    val runningGrade: Double? = null,
    val completedAt: String? = null,
    val credentialId: String? = null,
    val nextItem: IntroCourseNextItem? = null,
    val modules: List<IntroCourseModuleProgress>? = null,
    val welcomeBannerDismissed: Boolean? = null,
    val celebrationSeen: Boolean? = null,
)

enum class IntroCourseCardState {
    Hidden,
    Loading,
    Error,
    NotStarted,
    InProgress,
    Completed,
}